### PR TITLE
Improve demuxer code

### PR DESCRIPTION
**Pull request by Luni-4**
This PR removes a series of `TODO` present in the demuxer code and simplifies it a bit

### CODE REVIEW

**Comment by sdroege** [comment]
Seems good to me otherwise

File: src/demuxer.rs [hunk 1]
```diff
@@ -36,6 +38,22 @@ pub struct DemuxerParams {
     pub track_numbers: Option<Vec<u64>>,
 }
 
+enum TracType {
```
[the remark this answers is not in the document]
Comment (Luni-4) [reply src/demuxer.rs:41]: To fulfill this: `// TODO: Use an enum for the track type`

[the remark this answers is not in the document]
Comment (Luni-4) [reply src/demuxer.rs:41]: Ugh, typo :)

**Review by sdroege — APPROVED** [review]

File: src/demuxer.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (sdroege) [reply src/demuxer.rs]: Ok :) I thought there was maybe a reason, but I couldn't imagine one considering that Matroska doesn't use fourccs for everything

**Pull request merged**